### DISCUSSION TITLE
Can I reuse redisReply?

### DISCUSSION THREAD

**Discussion by hadjiprocopis** [redis/hiredis#1018] [Ideas]
Can I reuse a redisReply object so in order to improve efficiency of storing lots of data?

I am thinking something like:

```
redisReply *reply = NULL;
for(i=0;i<1000000;i++){
  redisCommand(redconn, &reply, "SET ...");
}
freeReplyObject(reply);
```

instead of:
```
redisReply *reply;
for(i=0;i<1000000;i++){
  reply = redisCommand(redconn, "SET ...");
  freeReplyObject(reply);
}
```
One way to implement this would be that each call to `redisCommand(redconn, &reply, "SET ...");`  checks if `*reply == NULL`, if it is then it malloc's it, if it isn't it just resets it and reuses it. Faster perhaps?

**Comment by michael-grunder** [comment]
HI :wave:

I'm honestly not exactly sure how you intend to reuse the reply.

redisCommand returns a redis reply:
```c
redisReply *reply = redisCommand(context, "SET %s %s", "foo", "bar");
```

So calling it as you do above wouldn't compile.

You can reuse replies as input to subsequent commands:

```c
reply1 = redisCommand(context, "GET %s", "somekey");
reply2 = redisCommand(context, "SET %s %s", "otherkey", reply->str);
```

It's not possible to re-use the malloc'd `redisReply` struct itself without changes to hiredis, but I would make sure this is actually a bottleneck.  I'm guessing the allocation of reply objects is much faster than the actual network communication to Redis itself.

**Comment by hadjiprocopis** [comment]
Thank you for looking at this. Investigating if it's a bottleneck first is a valid point.

thank you. (and thank you ALL for this nice package)
